## M20732/M342 
## Statistics Methods in Health Research & Social Science  
### Coursework (2022-23)  
---
### <font color=RoyalBlue> UP960303 </font>

### Instructions 
1. Your solutions must be submitted on Moodle no later than 23:55 pm on Friday 30 November 2022.
2. This coursework is divided into two parts, each worth 20% of the total mark for
this module. Answer all questions.
3. You must write up your work individually.
4. You must use R-Colab (or R studio) for one of the two parts.  
**<font color=RoyalBlue>Please write comments in your code and explain clearly what you are doing and the theoretical method to find the
solutions.</font>**
5. You need to show all steps in your calculations.
6. To get a good mark you must **<font color=RoyalBlue>justify every claim</font>**.
---

My comments will be denoted using comments within the code and with an overall markdown explanations in **<font color=RoyalBlue> blue</font>**.  
My findings from the statistical analysis will be in **<font color=Green> green </font>**.

This is where I have installed all of the libraries that I will need for the coursework so that they can run before the rest of the code.

In [19]:
library(tidyverse) 
library(haven)
library(epitools)
install.packages("readxl")
library("readxl")
install.packages("gmodels")
library(gmodels)
install.packages("samplesizeCMH")
library(samplesizeCMH)


The downloaded binary packages are in
	/var/folders/3m/hvvbkscs3g18gqtdpjbjc7l80000gn/T//Rtmplyoxxr/downloaded_packages

The downloaded binary packages are in
	/var/folders/3m/hvvbkscs3g18gqtdpjbjc7l80000gn/T//Rtmplyoxxr/downloaded_packages

The downloaded binary packages are in
	/var/folders/3m/hvvbkscs3g18gqtdpjbjc7l80000gn/T//Rtmplyoxxr/downloaded_packages


## Part 2
### These are the exercises that must be solved with R.
---

### Question D
A (fictitious) study was conducted to investigate the effect of cervical length, maternal characteristics and induced labour on the risk of a C-section delivery. Data were collected for 1028
women. Age, ethnicity, BMI and the number of previous pregnancies (parity) were among the
data collected. The Excel data file csw 2022 LogData.xlsx supplied for this assignment contains
the following information relating to:

In [20]:
# Here I will use the methods from previous lectures to abstract the columns into a table 
QuestionDData <- read_excel("CSW_2022_LogData.xlsx")
# I also need to rename columns in order to use then effectively within R 
colnames(QuestionDData)=c("PatientID","MA", "Ethnicity", "BMI","BMIgroup", "NoPreviousPregnancys", "FirstBaby", "CxLength", "CxGroup","Spontaneous","Caesarean" )
QuestionDData$Delivery <- QuestionDData$Caesarean
# Change the Data in Caesarean so that its a 1 or a 0 using the pipe operator
QuestionDData <- QuestionDData %>%
      mutate(Caesarean = ifelse(Caesarean == 1 ,0,1))
# Here im changing the data for First baby from 1/2 to yes/no
QuestionDData <- QuestionDData %>%
      mutate(FirstBaby = ifelse(FirstBaby == 1 ,'Yes', "No"))
# Here I am showing the top of the table 
head(QuestionDData)
# This table call is commented out but when in play shows the full table of data 
# QuestionDData

PatientID MA Ethnicity BMI  BMIgroup NoPreviousPregnancys FirstBaby CxLength
1 1         23 1         19.7 1        0                    Yes        4      
2 2         33 1         25.2 2        0                    Yes       14      
3 3         33 2         33.7 3        4                    No        10      
4 4         19 1         22.0 1        0                    Yes        7      
5 5         23 2         22.4 2        0                    Yes        7      
6 6         38 1         22.2 2        1                    No        14      
  CxGroup Spontaneous Caesarean Delivery
1 1       1           0         1       
2 1       1           1         2       
3 1       1           0         1       
4 1       1           0         1       
5 1       1           0         1       
6 1       1           0         1

1. [3 marks] Ignoring all other variables, determine whether there is evidence of an association between Caesarean delivery and BMI group. Interpret your finding.

<font color=RoyalBlue> Here I am creating a table for the results being tested. Then I run a Chi-Square test to determine whether there is an assosiation between having a Caesarean and BMI group </font>

In [21]:
TblCeaVSBMIG = table(QuestionDData$BMIgroup,QuestionDData$Caesarean)
colnames(TblCeaVSBMIG) <- c( "Vaginal", "Caesarean")
addmargins(TblCeaVSBMIG)

chisq.test(TblCeaVSBMIG ,correct = FALSE)


Vaginal Caesarean Sum 
  1   320      31        351
  2   271      64        335
  3   274      68        342
  Sum 865     163       1028


	Pearson's Chi-squared test

data:  TblCeaVSBMIG
X-squared = 19.787, df = 2, p-value = 5.051e-05


<font color=Green> The Chi Square statistic is 19.787, the P value is 0.000051. The p-value is less than 0.05 hence the result is significant. </font>

2. [3 marks] Obtain the odds ratios (and their CI) to describe the effect of BMI group on the
incidence of Caesarean delivery in this study population and interpret your result. Use the
first tertile (=1) as comparator group.

<font color=RoyalBlue> Code explanation here... </font>

In [22]:
oddsratio(TblCeaVSBMIG,correction=FALSE,verbose=TRUE,conf.level=0.05,method="wald")

$x
   
    Vaginal Caesarean
  1     320        31
  2     271        64
  3     274        68

$data
       
        Vaginal Caesarean Total
  1         320        31   351
  2         271        64   335
  3         274        68   342
  Total     865       163  1028

$p.exposed
       
          Vaginal Caesarean     Total
  1     0.3699422 0.1901840 0.3414397
  2     0.3132948 0.3926380 0.3258755
  3     0.3167630 0.4171779 0.3326848
  Total 1.0000000 1.0000000 1.0000000

$p.outcome
       
          Vaginal  Caesarean Total
  1     0.9116809 0.08831909     1
  2     0.8089552 0.19104478     1
  3     0.8011696 0.19883041     1
  Total 0.8414397 0.15856031     1

$measure
   odds ratio with 5% C.I.
    estimate    lower   upper
  1 1.000000       NA      NA
  2 2.437805 2.402314 2.47382
  3 2.561808 2.524838 2.59932

$conf.level
[1] 0.05

$p.value
   two-sided
      midp.exact fisher.exact   chi.square
  1           NA           NA           NA
  2 9.702354e-05 1.006172e-04 9.87302

3. [10 marks] Carry out a stratified analysis, using Firstbaby as first variable and Cx-group as the stratification variable.   
- Is there any evidence that Firstbaby has an effect on Delivery?   
- Is there any evidence that the risk due to Firstbaby is modified by the cervix size?   
- What are the odds ratio for Firstbaby (yes) after adjusting for Cx-group?

<font color=RoyalBlue> I need to extract the data for:   
- First Baby  
- Cx-Group  
- Delivery  
- Cervix Size  
 </font>

In [23]:
# First we need to seperate data into the different Cx Groups 
# Ill start by extracting  the data we need using aggregate
QDThreeData = aggregate(Caesarean ~ CxGroup + Delivery + FirstBaby , data = QuestionDData, FUN=length)
View(QDThreeData)

CxGroup Delivery FirstBaby Caesarean
1  1       1        No        156      
2  2       1        No        121      
3  3       1        No        108      
4  1       2        No          2      
5  2       2        No          6      
6  3       2        No         16      
7  1       1        Yes       196      
8  2       1        Yes       161      
9  3       1        Yes       123      
10 1       2        Yes        24      
11 2       2        Yes        37      
12 3       2        Yes        78

In [24]:
# Now I am creating a table out of the results that we need to evaluate
QDThreeTable = xtabs(Caesarean ~  FirstBaby + Delivery + CxGroup, data=QDThreeData)
ftable(QDThreeTable)

                   CxGroup   1   2   3
FirstBaby Delivery                    
No        1                156 121 108
          2                  2   6  16
Yes       1                196 161 123
          2                 24  37  78

In [25]:
# Now I'm finishing the tables by adding margins and spliting them by CxGroup

addmargins(QDThreeTable, margin = seq_along(dim(QDThreeTable)), FUN = sum, quiet = FALSE)

Margins computed over dimensions
in the following order:
1: FirstBaby
2: Delivery
3: CxGroup


, , CxGroup = 1

         Delivery
FirstBaby    1    2  sum
      No   156    2  158
      Yes  196   24  220
      sum  352   26  378

, , CxGroup = 2

         Delivery
FirstBaby    1    2  sum
      No   121    6  127
      Yes  161   37  198
      sum  282   43  325

, , CxGroup = 3

         Delivery
FirstBaby    1    2  sum
      No   108   16  124
      Yes  123   78  201
      sum  231   94  325

, , CxGroup = sum

         Delivery
FirstBaby    1    2  sum
      No   385   24  409
      Yes  480  139  619
      sum  865  163 1028


In [26]:
# Now we will analyse the data by peforming a CHI-Squared test 
apply(QDThreeTable, 3, chisq.test,correct = FALSE)

$`1`

	Pearson's Chi-squared test

data:  array(newX[, i], d.call, dn.call)
X-squared = 13.351, df = 1, p-value = 0.0002583


$`2`

	Pearson's Chi-squared test

data:  array(newX[, i], d.call, dn.call)
X-squared = 13.139, df = 1, p-value = 0.0002892


$`3`

	Pearson's Chi-squared test

data:  array(newX[, i], d.call, dn.call)
X-squared = 25.03, df = 1, p-value = 5.646e-07



Is there any evidence that Firstbaby has an effect on Delivery?
<font color=Green> 

I'm not super sure 

</font>

In [33]:
# Creating a table in order to be able to compare results that haven't been adjusted
QDThreeTableCompare = table(QuestionDData$FirstBaby, QuestionDData$Delivery ) 
QDThreeTableCompare

# Unadjusted Chi-Squared Test
CrossTable(QuestionDData$FirstBaby, QuestionDData$Delivery, expected = TRUE)


     
        1   2
  No  385  24
  Yes 480 139


 
   Cell Contents
|-------------------------|
|                       N |
|              Expected N |
| Chi-square contribution |
|           N / Row Total |
|           N / Col Total |
|         N / Table Total |
|-------------------------|

 
Total Observations in Table:  1028 

 
                        | QuestionDData$Delivery 
QuestionDData$FirstBaby |         1 |         2 | Row Total | 
------------------------|-----------|-----------|-----------|
                     No |       385 |        24 |       409 | 
                        |   344.149 |    64.851 |           | 
                        |     4.849 |    25.733 |           | 
                        |     0.941 |     0.059 |     0.398 | 
                        |     0.445 |     0.147 |           | 
                        |     0.375 |     0.023 |           | 
------------------------|-----------|-----------|-----------|
                    Yes |       480 |       139 |       619 | 
                        |   520.851 

Is there any evidence that the risk due to Firstbaby is modified by the cervix size?

<font color=Green> 

I'm not super sure 

</font>

<font color=RoyalBlue> In this code block I am running Odds Ratio on the 3 different CX-Groups and then I create a data table to run a odds ration for the total OR , then I run a odds ratio test without adjusting for CX-group. </font>

In [31]:
# Odds ratio for FirstBaby resulting in a casearean after adjusting for Cx-group
apply(QDThreeTable, 3, oddsratio.wald)
# Odds ratio for FirstBaby resulting in a Caesarean without adjusting for CX-group
oddsratio(QDThreeTableCompare,method="wald",conf.level=0.95)

$`1`
$`1`$data
         Delivery
FirstBaby   1  2 Total
    No    156  2   158
    Yes   196 24   220
    Total 352 26   378

$`1`$measure
         odds ratio with 95% C.I.
FirstBaby estimate    lower    upper
      No   1.00000       NA       NA
      Yes  9.55102 2.223106 41.03358

$`1`$p.value
         two-sided
FirstBaby   midp.exact fisher.exact   chi.square
      No            NA           NA           NA
      Yes 0.0001021056 0.0001256593 0.0002583262

$`1`$correction
[1] FALSE

attr(,"method")
[1] "Unconditional MLE & normal approximation (Wald) CI"

$`2`
$`2`$data
         Delivery
FirstBaby   1  2 Total
    No    121  6   127
    Yes   161 37   198
    Total 282 43   325

$`2`$measure
         odds ratio with 95% C.I.
FirstBaby estimate    lower   upper
      No  1.000000       NA      NA
      Yes 4.634576 1.895153 11.3338

$`2`$p.value
         two-sided
FirstBaby   midp.exact fisher.exact   chi.square
      No            NA           NA           NA
      Yes 0.0001583176 0.0001843496 0.0002892298

$`2`$correction
[1] FALSE

attr(,"method")
[1] "Unconditional MLE & normal approximation (Wald) CI"

$`3`
$`3`$data
         Delivery
FirstBaby   1  2 Total
    No    108 16   124
    Yes   123 78   201
    Total 231 94   325

$`3`$measure
         odds ratio with 95% C.I.
FirstBaby estimate    lower    upper
      No  1.000000       NA       NA
      Yes 4.280488 2.356769 7.774449

$`3`$p.value
         two-sided
FirstBaby   midp.exact fisher.exact   chi.square
      No            NA           NA           NA
      Yes 2.581796e-07 3.364403e-07 5.645874e-07

$`3`$correction
[1] FALSE

attr(,"method")
[1] "Unconditional MLE & normal approximation (Wald) CI"

$data
       
          1   2 Total
  No    385  24   409
  Yes   480 139   619
  Total 865 163  1028

$measure
     odds ratio with 95% C.I.
      estimate    lower    upper
  No  1.000000       NA       NA
  Yes 4.645399 2.951637 7.311107

$p.value
     two-sided
        midp.exact fisher.exact   chi.square
  No            NA           NA           NA
  Yes 6.106227e-14 7.312289e-14 1.028427e-12

$correction
[1] FALSE

attr(,"method")
[1] "Unconditional MLE & normal approximation (Wald) CI"

What are the odds ratio for Firstbaby (yes) after adjusting for Cx-group?
<font color=Green> 

This table shows all the results I've found  

|CX-Group | Risk Estimate| Value	| 95% CI |
|-------|--------------|--------|---------------------|
|1 |	OR for First Baby (yes/no)|	9.551 |	(2.223,	41.034) |
|2	| OR for First Baby (yes/no)|	4.635 |	(1.895,	11.334) |
|3	| OR for First Baby (yes/no)|	4.280  |	(2.357,	7.774)    |
|Total|	OR for First Baby (yes/no)|	4.645  |	(2.952,	7.311)   | 
  
From these numbers we can see that:   
- Members of CX-Group 1 are 9.6 times more likely to have a casesarean when its their first baby.
- Members of CX-Group 2 are 4.6 times more likely to have a casesarean when its their first baby.
- Members of CX-Group 3 are 4.3 times more likely to have a casesarean when its their first baby.

From these results we see that CX-group 1 appears to be more likely(in terms of odds) for having a Casesarean when its thier first birth than the other CX-Groups

</font>

4. [5 marks] Perform a multivariate analysis (logistic regression) to explore the relationship
between delivery and all seven of the potential predictor variables recorded (Do not use
group variables). (Hint: Your outcome of interest is Caesarean section yes or no, code the
outcome using a dummy variable in a suitable way).

<font color=RoyalBlue> Code explanation here... </font>

5. [3 marks] Find out and comment on the goodness of fit of your model.

<font color=RoyalBlue> Code explanation here... </font>

6. [10 marks] Describe the effect of each predictor on the likelihood of caesarean delivery.

<font color=RoyalBlue> Code explanation here... </font>

---
### Question E
Using the data from the randomised control trial comparing treatments for ulcerative colitis  
(Question C):  

| Randomised Control Trial |   |   |   |    |   |    |    |     |     |    |     |    |     |     |    |     |     |     |     |     |     |     |     |
|--------------------------|---|---|---|----|---|----|----|-----|-----|----|-----|----|-----|-----|----|-----|-----|-----|-----|-----|-----|-----|-----|
| Test group:              | 2 | 6 | 6 | 6+ | 7 | 9+ | 10 | 10+ | 11+ | 13 | 17+ | 18 | 19+ | 20+ | 22 | 24+ | 25+ | 32+ | 32+ | 34+ | 35+ | 36+ |     |
| Control group:           | 1 | 2 | 2 | 3  | 4 | 4  | 5  | 5   | 8   | 8  | 10  | 11 | 11+ | 12  | 14 | 15  | 16+ | 19  | 24  | 27  | 29+ | 31  | 36+ |


  
**Answer the following questions.**

1. [1 marks] Enter the data for both the test and control groups into R and code the variables
appropriately.

<font color=RoyalBlue> Code explanation here... </font>

2. [3 marks] Produce graphs of the Kaplan Meier estimate of the survival functions, S(t).
What do these graphs suggest?

<font color=RoyalBlue> Code explanation here... </font>

3. [3 marks] Produce a plot of the cumulative hazard functions. What do these graphs
suggest?

<font color=RoyalBlue> Code explanation here... </font>

4. [3 marks] For the control group, what is the (K-M) estimated probability of survival beyond
12 weeks? Ignoring the small sample size, give approximate 95% confidence limits for the
true value.

<font color=RoyalBlue> Code explanation here... </font>

5. [3 marks] Carry out a Logrank test to determine if there is evidence of a significant difference
in survival functions. What do you conclude?

<font color=RoyalBlue> Code explanation here... </font>

6. [3 marks] Fit a Cox regression model to these data using treatment as a single predictor and
estimate the hazard ratio for the test treatment together with a 95% confidence interval.
Interpret your result.

<font color=RoyalBlue> Code explanation here... </font>